File: solutions/_380_insert_delete_getrandom_o1/solution.java
package _380_insert_delete_getrandom_o1;

import java.util.*;

class RandomizedSet {

  private Set<Integer> set;

  public RandomizedSet() {
    this.set = new HashSet<>();
  }

  public boolean insert(int val) {
    if (this.set.contains(val)) {
      return false;
    }
    this.set.add(val);
    return true;
  }

  public boolean remove(int val) {
    if (!this.set.contains(val)) {
      return false;
    }
    this.set.remove(val);
    return true;
  }

  public int getRandom() {
    int index = new Random().nextInt(this.set.size());
    return (int) this.set.toArray()[index];
  }
}
/*
https://leetcode.com/problems/insert-delete-getrandom-o1/
*/

File: solutions/_380_insert_delete_getrandom_o1/solution.js
class RandomizedSet {
  constructor() {
    this.set = new Set();
  }

  insert(val) {
    if (this.set.has(val)) {
      return false;
    }
    this.set.add(val);
    return true;
  }

  remove(val) {
    if (!this.set.has(val)) {
      return false;
    }
    this.set.delete(val);
    return true;
  }

  getRandom() {
    const randomIndex = Math.floor(Math.random() * this.set.size);
    const iterator = this.set.values();
    for (let i = 0; i < randomIndex; i++) {
      iterator.next();
    }
    return iterator.next().value;
  }
}

module.exports = { RandomizedSet };

/*
https://leetcode.com/problems/insert-delete-getrandom-o1/
*/
